pkg/ipamd/ipamd.go

```diff
@@ -1058,6 +1056,10 @@ func (c *IPAMContext) shouldRemoveExtraENIs() bool {
 	if shouldRemoveExtra {
 		logPoolStats(total, used, c.maxIPsPerENI, c.enableIpv4PrefixDelegation)
 		log.Debugf("It might be possible to remove extra ENIs because available (%d) >= (ENI/Prefix target + 1 (%d) + 1) * addrsPerENI (%d)", available, warmTarget, c.maxIPsPerENI)
+	} else if c.enableIpv4PrefixDelegation {
+		// When prefix target count is reduced, datastorehigh would have deleted extra prefixes hence available will be less than prefix target
```

Comment:
It should only delete extra prefixes that we no longer need to meet the reduced prefix target(?)..Why will this delete more prefixes than what is required to maintain WARM_PREFIX_TARGET?

Reply:
Oh I just need to reword the comment. So DatastoreHigh check would have deleted prefixes which are more than WARM_PREFIX_TARGET and these deleted prefixed might be spread across ENIs and those ENIs will be deletable now. So `shouldRemoveExtraENIs ` should return true now so that `tryFreeENI` frees it up.